File: plugins/rd/skills/10-stages-developing/templates/typescript/function-template.ts
/**
 * [Brief description of what this module does]
 */

/**
 * [Brief description of what the function does].
 *
 * @param param1 - [Description of param1]
 * @param param2 - [Description of param2]
 * @returns Object containing processed results
 * @throws {Error} If param1 is empty or param2 is negative
 *
 * @example
 * const result = functionName('example', 42);
 * // Returns: { key1: 'processed_example', key2: 84 }
 */
export function functionName(param1: string, param2: number): { key1: string; key2: number } {
    // Input validation
    if (!param1) {
        throw new Error('param1 cannot be empty');
    }

    if (param2 < 0) {
        throw new Error('param2 must be non-negative');
    }

    // Implementation
    const result = {
        key1: `processed_${param1}`,
        key2: param2 * 2
    };

    return result;
}

File: plugins/rd/skills/10-stages-developing/templates/go/function-template.go
// Package module provides [brief description of what this module does]
package module

import (
	"errors"
	"fmt"
)

// Result represents the result of FunctionName
type Result struct {
	Key1 string
	Key2 int
}

// FunctionName [brief description of what the function does]
//
// Parameters:
//   - param1: [Description of param1]
//   - param2: [Description of param2]
//
// Returns:
//   - Result struct containing processed results
//   - error if validation fails
//
// Example:
//   result, err := FunctionName("example", 42)
//   // Returns: Result{Key1: "processed_example", Key2: 84}, nil
func FunctionName(param1 string, param2 int) (Result, error) {
	// Input validation
	if param1 == "" {
		return Result{}, errors.New("param1 cannot be empty")
	}

	if param2 < 0 {
		return Result{}, errors.New("param2 must be non-negative")
	}

	// Implementation
	result := Result{
		Key1: fmt.Sprintf("processed_%s", param1),
		Key2: param2 * 2,
	}

	return result, nil
}
